examples/trading/api/server.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import express from 'express';
+
+import { prisma } from './db';
+import {
+	formatPaginatedResponse,
+	parsePaginationForQuery,
+	parseWhereStatement,
+	WhereParam,
+	WhereParamTypes,
+} from './utils/api-queries';
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/', async (req, res) => {
+	return res.send({ message: '🚀 API is functional 🚀' });
+});
+
+app.get('/locked', async (req, res) => {
+	const acceptedQueries: WhereParam[] = [
+		{
+			key: 'deleted',
+			type: WhereParamTypes.BOOLEAN,
+		},
+		{
+			key: 'creator',
+			type: WhereParamTypes.STRING,
+		},
+		{
+			key: 'keyId',
+			type: WhereParamTypes.STRING,
+		},
+	];
+
+	try {
+		const locked = await prisma.locked.findMany({
+			where: parseWhereStatement(req.query, acceptedQueries)!,
+			...parsePaginationForQuery(req.query),
+		});
+
+		return res.send(formatPaginatedResponse(locked));
+	} catch (e) {
+		console.error(e);
+		return res.status(400).send(e);
+	}
+});
+
+app.get('/escrows', async (req, res) => {
+	const acceptedQueries: WhereParam[] = [
+		{
+			key: 'cancelled',
+			type: WhereParamTypes.BOOLEAN,
+		},
+		{
+			key: 'swapped',
+			type: WhereParamTypes.BOOLEAN,
+		},
+		{
+			key: 'recipient',
+			type: WhereParamTypes.STRING,
+		},
+		{
+			key: 'sender',
+			type: WhereParamTypes.STRING,
+		},
+	];
+
+	try {
+		const escrows = await prisma.escrow.findMany({
+			where: parseWhereStatement(req.query, acceptedQueries)!,
+			...parsePaginationForQuery(req.query),
+		});
+
+		return res.send(formatPaginatedResponse(escrows));
+	} catch (e) {
+		console.error(e);
+		return res.status(400).send(e);
+	}
+});
+
+app.listen(3000, () =>
+	console.log(`
```

Comment:
nit: what's with the newline here?

Reply:
That comes from the global (monorepo) linter, probably based on a max width setting.

Reply:
I meant within the string here -- I would be really suprised if a linter changed the string output!